src/mqtt/mqtt_test.c

```diff
@@ -1166,17 +1388,57 @@ TEST( MqttTest, MQTT_Publish_With_Retain_Flag )
                                                     MQTT_GetPacketId( &context ) ) );
     /* Complete the QoS 1 PUBLISH operation. */
     TEST_ASSERT_FALSE( receivedPubAck );
-    TEST_ASSERT_EQUAL( MQTTSuccess,
-                       MQTT_ProcessLoop( &context, 2 * MQTT_PROCESS_LOOP_TIMEOUT_MS ) );
+    
+    entryTime = FRTest_GetTimeMs();
+    do
+    {
+        xMQTTStatus = MQTT_ProcessLoop( &context );
+
+        if( FRTest_GetTimeMs() > ( entryTime + ( MQTT_PROCESS_LOOP_TIMEOUT_MS * 2 ) ) )
+        {
+            /* Timeout. */
+            break;
+        }
+        else
+        {
+            /* Sleep for some time between iterations. */
+            FRTest_TimeDelay( MQTT_PROCESS_LOOP_TIMEOUT_MS / 50 );
```

Comment:
why 50 here? your delays both divide by 50 - should this number be configurable by a macro?

Reply:
Done!